src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -101,16 +127,75 @@ export default class VizShonenJump extends Connector {
             });
     }
 
+    async _getMangaVolumes(manga) {
+        let auth = await this._getUserInfo();
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        return [...data.querySelectorAll('#o_products tr td:last-of-type a')]
+            .filter(element => {
+                // login required (e.g. age verification) => javascript:void('login to read');
+                if(/(?=javascript:)(?=.*login)/i.test(element.href)) {
+                    return auth.isLoggedIn && auth.isAdult;
+                }
+                // subscription required => javascript:void('join to read');
+                if(/(?=javascript:)(?=.*join)/i.test(element.href)) {
+                    return auth.isMember;
+                }
+                // free
+                return true;
+            })
+            .map(chapter => {
+                return {
+                    id: chapter.href.replace("hakuneko://cache/", ""),
+                    title: 'Vol. ' + chapter.href.match(/-volume-([-_0-9]+)/)[1]
+                };
+            });
+    }
+
+    async _getChapters(manga) {
+        if (manga.id.startsWith("/shonenjump/chapters")) {
+            return await this._getMangaChapters(manga);
+        } else if (manga.id.startsWith("/library")) {
+            return await this._getMangaVolumes(manga);
+        }
+    }
+
+    async requestRegex(responseData, regex) {
+        let regex2 = new RegExp(regex.source, regex.flags + "g");
+        let result = [];
+        let match = undefined;
+        // eslint-disable-next-line no-cond-assign
+        while(match = regex2.exec(responseData)) {
+            result.push(match[1]);
+        }
+
+        return result;
+    }
+
     async _getPages(chapter) {
         const request = new Request(new URL(chapter.id, this.url));
-        const data = await this.fetchRegex(request, /var\s+pages\s*=\s*(\d+)/g);
-        const pageCount = parseInt(data[0]);
-        const chapterID = chapter.id.match(/chapter\/(\d+)/)[1];
+        let response = await fetch(request);
+        let responseData = await response.text();
+
+        let pageCount = 0;
+        let mangaID = 0;
+
+        if (chapter.id.startsWith("read/manga")) {
+            const pageCountData = await this.requestRegex(responseData, /<div class="mar-b-md"><strong>Length<\/strong> (\d+) pages<\/div>/);
```

Comment:
The [slightest change with whitespaces](https://regex101.com/r/YflMBl/1/) (linebreak, space) will cause this to fail, isn't there a more reliable way to get the page count?

Reply:
Unfortunately not as far as I can tell, I can send an example page to you if you'd like to take a look and see if a better regex could be constructed for this. I'm not much of a wiz at these as you can tell, mostly just built off what I saw already here.

Reply:
Filling in whitespace may be sufficient
`/<div\s+class="mar-b-md">\s*<strong>\s*Length\s*<\/strong>\s+(\d+)\s+pages\s*<\/div>/`